src/esp/scene/SceneNode.cpp

```diff
@@ -12,26 +14,173 @@ namespace scene {
 
 SceneNode::SceneNode(SceneNode& parent)
```

Comment:
Hold on here:
You have code duplication in the constructors.
You need constructor delegation.